### PR TITLE
WIP: Add compat for new AbstractCell type.

### PR DESCRIPTION
**Pull request by fredrikekre**

### CODE REVIEW

**Comment by fredrikekre** [comment]
I think the new interface will be a win in FerriteViz too, most dispatches seem to be based on the reference shape, which you can now to directly instead of figuring out the correct numbers in `Cell{refdim, nnodes, nfaces}`.

**Comment by termi-official** [comment]
I introduced the abstract methods for now to have at least some kind of viz. Actually the high order stuff needs quite a different treatment but I could not find time yet.

File: src/lor_tools.jl [hunk 1]
```diff
@@ -10,65 +10,65 @@ for_nodes(::Union{Ferrite.Lagrange{2,Ferrite.RefTetrahedron,2},Ferrite.Discontin
     (3,6,5),
     (5,4,2),
 )
-# Cubic Triangle
-for_nodes(::Union{Ferrite.Lagrange{2,Ferrite.RefTetrahedron,3},Ferrite.DiscontinuousLagrange{2,Ferrite.RefTetrahedron,3},Ferrite.Cell{2,10,3}}) = (
```
Comment (termi-official) [inline src/lor_tools.jl:14]: Note that this also includes the refinement for interpolations. Maybe we just want to remove the Cell from the type union for now. Same for ones below.

File: src/utils.jl [hunk 2]
```diff
@@ -177,25 +205,24 @@ num_vertices(p::MakiePlotter) = size(p.physical_coords,1)
...
 """
 # Helper to count triangles e.g. for preallocations.
-ntriangles(cell::Ferrite.AbstractCell{2,3,3}) = 1 # Tris in 2D
-ntriangles(cell::Ferrite.AbstractCell{3,3,1}) = 1 # Tris in 3D
-ntriangles(cell::Ferrite.AbstractCell{dim,N,4}) where {dim,N} = 4 # Quads in 2D and 3D
-ntriangles(cell::Ferrite.AbstractCell{3,N,1}) where N = 4 # Tets as a special case of a Quad, obviously :)
-ntriangles(cell::Ferrite.AbstractCell{3,N,6}) where N = 6*4 # Hex
+ntriangles(cell::TriangleBasedCells)      = 1 # Tris in 2D and 3D
+ntriangles(cell::QuadrilateralBasedCells) = 4 # Quads in 2D and 3D
+# ntriangles(cell::Ferrite.AbstractCell{3,N,1}) where N = 4 # Tets as a special case of a Quad, obviously :)
```
Comment (termi-official) [inline src/utils.jl:210]: Can be removed now that embedded cells do not have the case distinction anymore.

```suggestion
```

**Review by termi-official — CHANGES_REQUESTED** [review]
LGTM otherwise. Fixing the first one should make the CI happy (fails because the first order refinement for interpolations is missing).

**Comment by fredrikekre** [comment]
I am donating this PR to someone who knows better :) The current changes are mostly mechanical without knowing much how these things are used.

**Comment by termi-official** [comment]
Will take over then.

**Comment by termi-official** [comment]
Handled in #103

**Pull request closed without merging**